polaris-react/src/components/TextField/TextField.tsx

```diff
@@ -591,7 +591,19 @@ export function TextField({
     setFocus(true);
   }
 
-  function handleBlur() {
+  function withinInput(target: HTMLElement | EventTarget) {
+    return (
+      target instanceof HTMLElement &&
+      verticalContentRef.current &&
+      verticalContentRef.current.contains(target)
+    );
+  }
+
+  function handleBlur({target}: React.FocusEvent) {
+    if (withinInput(target) && focus) {
```

Comment:
This prevents unwanted focus loss when interacting with nodes inside of the `verticalContent` when provided.

Reply:
I think this might be causing some unintended behavior where focus remains when trying to click outside the textfield. I tested the snapshot on my spin instance and I'm seeing this behavior:

https://user-images.githubusercontent.com/26749317/170699475-30ab8255-43c8-4780-88d0-1487a3b0f665.mp4

Reply:
Since `onBlur` and `onFocus` event targets only return the node that is listening for the event (no `currentTarget` or `relatedTarget`) unless the focus was moved to another focusable element (like a button), there's no way to determine whether or not the blur of the input is caused by selecting the text of applied tags at the parent level. I'm going to reapproach fixing this by reworking the focus management of `TextField` this afternoon 🤞🏽

Reply:
I think I've got a working fix!! Gonna update my snapshot and link to my spinstance for tophatting 🎩